File: go/pkg/limits.go
package pkg

const MultimapElemCountLimit = 1024

const FixedHdrContentSizeLimit = 1 << 20
const VarHdrContentSizeLimit = 1 << 20

// RecordAllocLimit defines the maximum number of bytes that can be allocated
// during the reading of a single record.
const RecordAllocLimit = 1 << 25 // 32 MiB

// FrameSizeLimit is the maximum allowed uncompressed or compressed size for a frame.
const FrameSizeLimit = 1 << 26

File: rust/stef-core/src/limits.rs
/// Maximum number of elements in a multimap field during decoding.
pub const MULTIMAP_ELEM_COUNT_LIMIT: u64 = 1024;

/// Maximum fixed-header content size.
pub const FIXED_HDR_CONTENT_SIZE_LIMIT: u64 = 1 << 20;

/// Maximum variable-header content size.
pub const VAR_HDR_CONTENT_SIZE_LIMIT: u64 = 1 << 20;

/// Maximum bytes allocated while decoding one record.
pub const RECORD_ALLOC_LIMIT: usize = 1 << 25;

/// Maximum compressed or uncompressed frame size.
pub const FRAME_SIZE_LIMIT: u64 = 1 << 26;
